src/sign/participant.rs

```diff
@@ -208,6 +240,67 @@ impl InnerProtocolParticipant for SignParticipant {
     }
 
     fn set_ready(&mut self) {
-        self.ready = true;
+        if self.status == Status::NotReady {
+            self.status = Status::Initialized;
+        } else {
+            warn!(
+                "Something is strange in the status updates for signing.
+                 Tried to update from `NotReady` to `Initialized`, but status was {:?}",
+                self.status
+            )
```

Comment:
again, not a fully-fledged state machine here but at least it'll be logged if we're having state mismatch problems.